src/flamegraph/color.rs

```diff
@@ -0,0 +1,272 @@
+use std::str::FromStr;
+use std::collections::HashMap;
+use std::fs::OpenOptions;
+use std::io;
+use std::io::Write;
+use std::path::Path;
+use std::fs::File;
+use std::io::BufReader;
+use std::io::BufRead;
+
+pub(super) const VDGREY: &str = "rgb(160,160,160)";
+pub(super) const DGREY: &str = "rgb(200,200,200)";
+
+const YELLOW_GRADIENT: (&str, &str) = ("#eeeeee", "#eeeeb0");
+const BLUE_GRADIENT: (&str, &str) = ("#eeeeee", "#e0e0ff");
+const GRAY_GRADIENT: (&str, &str) = ("#f8f8f8", "#e8e8e8");
+
+#[derive(Debug, PartialEq)]
+pub enum Palette {
+    Hot,
+    Mem,
+    Io,
+    Wakeup,
+    Chain,
+    Java,
+    Js,
+    Perl,
+    Red,
+    Green,
+    Blue,
+    Aqua,
+    Yellow,
+    Purple,
+    Orange,
+}
+
+impl Default for Palette {
+    fn default() -> Self {
+        Palette::Hot
+    }
+}
+
+impl FromStr for Palette {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hot" => Ok(Palette::Hot),
+            "mem" => Ok(Palette::Mem),
+            "io" => Ok(Palette::Io),
+            "wakeup" => Ok(Palette::Wakeup),
+            "chain" => Ok(Palette::Chain),
+            "java" => Ok(Palette::Java),
+            "js" => Ok(Palette::Js),
+            "perl" => Ok(Palette::Perl),
+            "red" => Ok(Palette::Red),
+            "green" => Ok(Palette::Green),
+            "blue" => Ok(Palette::Blue),
+            "aqua" => Ok(Palette::Aqua),
+            "yellow" => Ok(Palette::Yellow),
+            "purple" => Ok(Palette::Purple),
+            "orange" => Ok(Palette::Orange),
+            unknown=> Err(format!("unknown color palette: {}", unknown))
+        }
+    }
+}
+
+fn namehash(name: &str) -> f32 {
+    let mut vector = 0f32;
```

Comment:
I would just use `.0` (so `0.0`) for these and let the compiler choose the type. That way moving to something like `f64` later would be an easier change.